module/motion/IKKick/src/IKKick.cpp

```diff
@@ -133,18 +133,20 @@ namespace motion {
                 Eigen::Affine3d torsoPose =
                     (supportFoot == LimbID::LEFT_LEG) ? leftFoot.inverse() : rightFoot.inverse();
 
-                // Put the ball position from vision into torso coordinates
-                Eigen::Vector3d targetTorso;  // = Transform3D(convert<double, 4,
-                                              // 4>(sensors.kinematicsBodyToGround)).i().transformPoint(convert<double,
-                                              // 3>(command.target)); //TODO fix
+                Eigen::Affine3d kinematicsGroundToBody = Eigen::Affine3d(sensors.kinematicsBodyToGround).inverse();
+
+                // Create the command target point object so we can transform it
+                Eigen::Vector4d commandTargetPoint(command.target[0], command.target[1], command.target[2], 1);
+
+                // Put the ball position from vision into torso coordinates by transforming the command target point.
+                Eigen::Vector3d targetTorso = (kinematicsGroundToBody * commandTargetPoint).block<3, 1>(0, 0);
+
                 // Put the ball position into support foot coordinates
                 Eigen::Vector3d targetSupportFoot = torsoPose * targetTorso;
 
                 // Put the goal from vision into torso coordinates
-                Eigen::Vector3d
-                    directionTorso;  // = Transform3D(convert<double, 4,
-                                     // 4>(sensors.kinematicsBodyToGround)).i().transformVector(convert<double,
-                                     // 3>(command.direction)); //TODO fix
+                Eigen::Vector3d directionTorso(kinematicsGroundToBody * command.direction);
```

Comment:
`command.direction` is `vec3`. This code is invalid.

Reply:
I am pretty sure it's valid. based on [the eigen docs](https://eigen.tuxfamily.org/dox/classEigen_1_1Transform.html):
> A little exception to the "as pure matrix product" rule is the case of the transformation of non homogeneous vectors by an affine transformation. In that case the last matrix row can be ignored, and the product returns non homogeneous vectors.

If I am understanding correctly, multiplying a homogeneous transform with a non-homogeneous vector does equivalent operations to what `Transform3D::transformVector` does, which is just multiply the vector by the rotation matrix, because it doesn't make sense for a vector to have translation.

Unless you're saying I need to explicitly cast the `vec3` message type into a `Vector3d`, in which case I would have thought that the compiler would complain?

Reply:
Ah ok, nevermind then